### DISCUSSION TITLE
What should we expect from argo workflow v3?

### DISCUSSION THREAD

**Discussion by Bobgy** [argoproj/argo-workflows#4953] [Q&A] [answered]
I can see there are some items in https://github.com/argoproj/argo/milestone/20 with very brief explanation.

I'd like to know some general questions:
* Are there going to be breaking changes? If yes, which are they?
* Are you on track? any updated ETA? the DDL for that milestone seems to be tomorrow

I'm asking these in the context of https://github.com/kubeflow/pipelines/issues/4553.
Would you recommend direct upgrade to argo v3, or stay in argo v2.12 LTS for a while?

**Comment by alexec** [comment, marked as answer]
We'll be publishing more details in about a week. Hopefully RC-1 within the week, and GA maybe in 4 weeks.

We don't expect breaking changes for most users.

For Kubeflow, I'd recommend going to v2.12 as the jump from v2.4 to v2.12 is much larger than v2.12 to v3.0.

**Reply by Bobgy** [reply]
Understood, thank you for the quick reply!

**Comment by juliusvonkohout** [comment]
Actually it would be very good to have the emissary executor for runasnonroot pipelines https://github.com/argoproj/argo/blob/22f4d26ec059ad86b9c409ffe25392261a22afba/docs/workflow-executors.md#emissary-emissary . In which versions is it included?
